### PR TITLE
Adding PublicNetworkAccess, InfrastructureEncryption, ResourceIdentity as Server Create/Update property for PostgreSQL/MySQL/MariaDB

### PR DESCRIPTION
**Pull request by ramnov**
Adding PublicNetworkAccess, InfrastructureEncryption, ResourceIdentity as Server Create/Update property for PostgreSQL/MySQL/MariaDB

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by lirenhe — APPROVED** [review]

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/stable/2018-06-01/mariadb.json [hunk 1]
```diff
@@ -1572,6 +1576,10 @@
           "$ref": "#/definitions/SslEnforcement",
           "description": "Enable ssl enforcement or not when connect to server."
         },
+        "publicNetworkAccess": {
```
Comment (anthony-c-martin) [inline specification/mariadb/resource-manager/Microsoft.DBforMariaDB/stable/2018-06-01/mariadb.json]: This is typically regarded as a breaking change to a GA API. Why not publish a new API version?

File: specification/mariadb/resource-manager/Microsoft.DBforMariaDB/stable/2018-06-01/mariadb.json [hunk 2]
```diff
@@ -1786,6 +1794,10 @@
               "$ref": "#/definitions/SslEnforcement",
               "description": "Enable ssl enforcement or not when connect to server."
             },
+            "publicNetworkAccess": {
```
Comment (anthony-c-martin) [inline specification/mariadb/resource-manager/Microsoft.DBforMariaDB/stable/2018-06-01/mariadb.json]: This is typically regarded as a breaking change to a GA API. Why not publish a new API version?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json [hunk 3]
```diff
@@ -1796,6 +1804,14 @@
           "$ref": "#/definitions/MinimalTlsVersion",
           "description": "Enforce a minimal Tls version for the server."
         },
+        "infrastructureEncryption": {
```
Comment (anthony-c-martin) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json]: This looks again like a breaking change.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json [hunk 4]
```diff
@@ -1991,6 +2007,10 @@
     },
     "ServerForCreate": {
       "properties": {
+        "identity": {
```
Comment (anthony-c-martin) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json]: Possible breaking change.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json [hunk 5]
```diff
@@ -2020,6 +2040,10 @@
     },
     "ServerUpdateParameters": {
       "properties": {
+        "identity": {
```
Comment (anthony-c-martin) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json]: Possible breaking change.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json [hunk 6]
```diff
@@ -2047,6 +2071,10 @@
               "$ref": "#/definitions/MinimalTlsVersion",
               "description": "Enforce a minimal Tls version for the server."
             },
+            "publicNetworkAccess": {
```
Comment (anthony-c-martin) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json]: Possible breaking change.

**Review by anthony-c-martin — CHANGES_REQUESTED** [review]
There are multiple breaking changes in this PR. What is the motivation?

**Comment by anthony-c-martin** [comment]
Discussed offline - these changes have already been partially signed off on, and we're adding properties that were previously missed.

**Review by anthony-c-martin — APPROVED** [review]

**Pull request merged**